### PR TITLE
Accept only TLS1.2 and TLs1.3 #624

### LINKED ISSUE #624: Disable TLS 1.1 and TLS 1.2
We want to disable TLS 1.0 and TLS 1.1 for security reasons

Links:
- https://stackoverflow.com/questions/38302683/how-can-i-disable-tlsv1-0-with-spring-boot-and-embedded-tomcat
- https://www.baeldung.com/spring-tls-setup

### LINKED ISSUE #625: Disable SHA1 possiblities for TLS 1.2
We want to disable the possiblity to use SHA1 in TLS 1.2 (TLS 1.3 does not support it any longer).

_This should be possible by changing spring boot configuration_

We will setup dedicated ciphers for
- SecHub Server
- PDS

### LINKED ISSUE #645: Disable TLS 1.1 and TLS 1.2 - for PDS
same as #624 but for PDS server and not SecHub server

### PR DESCRIPTION
**Pull request by de-jcup**
Security improvement PR

- closes #624 
- closes #645 
- closes #625

### CODE REVIEW

File: sechub-integrationtest/src/test/java/com/daimler/sechub/integrationtest/security/ServerEncryptionTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+class ServerEncryptionTest {
+
+    private static SecurityTestHelper securityTestHelper;
+
+    @BeforeAll
+    static void beforeAll() throws Exception {
+        IntegrationTestContext context = IntegrationTestContext.get();
+
+        String checkAlive = context.getUrlBuilder().buildCheckIsAliveUrl();
+        securityTestHelper = new SecurityTestHelper(TestTargetType.SECHUB_SERVER,new URL(checkAlive));
```
Comment (Jeeppler) [inline sechub-integrationtest/src/test/java/com/daimler/sechub/integrationtest/security/ServerEncryptionTest.java:26]:
```suggestion
        securityTestHelper = new SecurityTestHelper(TestTargetType.SECHUB_SERVER, new URL(checkAlive));
```

**Review by Jeeppler — CHANGES_REQUESTED** [review]
See comments.

File: sechub-integrationtest/ciphertest.sh [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+OUTPUT_FILE="$OUTPUT_FOLDER/sechub-$2.json"
+
+rm -f $OUTPUT_FILE
+mkdir -p $OUTPUT_FOLDER
+
+echo DEV_CERT_FILE=$DEV_CERT_FILE
+
+SERVER=$1
+
+ciphers=$(openssl ciphers 'ALL:eNULL' | sed -e 's/:/ /g')
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-integrationtest/ciphertest.sh:46]: done

File: sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/SecurityTestHelper.java [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+    public void assertProtocolAccepted(String protocol) throws Exception {
+        SSLTestContext context = new SSLTestContext();
+        context.protocol = protocol;
+        context.expectProtocolNotAccepted = false;
+
+        callTestURLWithProtocol(context);
+    }
+    
+    String getMac(CipherCheck check) {
+        String cipher  =check.cipher;
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/SecurityTestHelper.java:86]: done - sourcecode formatter applied

File: sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/SecurityTestHelper.java [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+        if (cipher==null) {
+            return null;
+        }
+        int lastIndex = cipher.lastIndexOf("-");
+        if (lastIndex==-1) {
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/SecurityTestHelper.java:91]: done - sourcecode formatter applied

File: sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/SecurityTestHelper.java [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+                String mac = getMac(check);
+                
+                if (mac==null) {
+                    fail("mac is null in test cipher - should not happen!");
+                }
+                
+                for (String notAllowedMac: notAllowedMacs) {
+                    
+                    if (mac.equalsIgnoreCase(notAllowedMac)){
+                        fail("Not wanted mac:"+mac+" found inside verfified cipher:"+check.cipher);
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/SecurityTestHelper.java:227]: done

File: sechub-integrationtest/src/test/java/com/daimler/sechub/integrationtest/security/ServerEncryptionTest.java (hunk 1 above)
Comment (de-jcup) [reply sechub-integrationtest/src/test/java/com/daimler/sechub/integrationtest/security/ServerEncryptionTest.java:26]: done - sourcecode formatter applied

**Review by Jeeppler — APPROVED** [review]

**Pull request merged**